src/lib.rs

```diff
@@ -1082,7 +1082,7 @@ impl<'a, X, F> de::MapVisitor for MapVisitor<'a, X, F>
             parent: self.path,
             key: self.key()?,
         };
-        self.delegate.visit_value_seed(TrackedSeed::new(seed, &mut self.callback, path))
+        self.delegate.visit_value_seed(TrackedSeed::new(seed, &mut *self.callback, path))
```

Comment:
This specifically was the line that crucially needed changing (I found at least).

I tried to keep `&'b mut F` out of `Deserializer` itself but it ended up needing to get pushed that high up (I think)